clustered/common/src/main/java/org/ehcache/clustered/common/messages/EhcacheCodec.java

```diff
@@ -34,30 +34,28 @@
 
   @Override
   public byte[] encodeMessage(EhcacheEntityMessage message) {
-    if (message.getType() == EhcacheEntityMessage.Type.LIFECYCLE_OP) {
-      return LifeCycleOpCodec.encode((LifecycleMessage) message);
-    } else {
-      return ServerStoreOpCodec.encode((ServerStoreOpMessage) message);
-    }
+    return message.encode();
```

Comment:
The problem in this context is that we may end up having to support evolution in messages to guarantee backward compatibility. Having the logic isolated in codec classes makes more sense in that context.
Could others chime in?